indexing-service/src/main/java/io/druid/indexing/overlord/ThreadPoolTaskRunner.java

```diff
@@ -240,85 +238,44 @@ public void stop()
   {
     final TaskToolbox toolbox = toolboxFactory.build(task);
     final Object taskPriorityObj = task.getContextValue(TaskThreadPriority.CONTEXT_KEY);
-    int taskPriority = 0;
-    if (taskPriorityObj != null) {
-      if (taskPriorityObj instanceof Number) {
-        taskPriority = ((Number) taskPriorityObj).intValue();
-      } else if (taskPriorityObj instanceof String) {
-        try {
-          taskPriority = Integer.parseInt(taskPriorityObj.toString());
-        }
-        catch (NumberFormatException e) {
-          log.error(e, "Error parsing task priority [%s] for task [%s]", taskPriorityObj, task.getId());
-        }
-      }
-    }
+    final int taskPriority = taskPriorityObj == null ? 0 : Numbers.parseInt(taskPriorityObj);
```

Comment:
The behavior here is different - previously if the priority was unparseable we would log an error and ignore it. Is that intentional?

Reply:
Thanks. Fixed.